src/PowerShellEditorServices/Services/PowerShell/Handlers/EvaluateHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Management.Automation;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell.Execution;
+
+namespace Microsoft.PowerShell.EditorServices.Handlers
+{
+    internal class EvaluateHandler : IEvaluateHandler
+    {
+        private readonly ILogger _logger;
+        private readonly PowerShellExecutionService _executionService;
+
+        public EvaluateHandler(
+            ILoggerFactory factory,
+            PowerShellExecutionService executionService)
+        {
+            _logger = factory.CreateLogger<EvaluateHandler>();
+            _executionService = executionService;
+        }
+
+        public Task<EvaluateResponseBody> Handle(EvaluateRequestArguments request, CancellationToken cancellationToken)
+        {
+            _executionService.ExecutePSCommandAsync(
+                new PSCommand().AddScript(request.Expression),
+                new PowerShellExecutionOptions { WriteInputToHost = true, WriteOutputToHost = true, AddToHistory = true, InterruptCommandPrompt = true },
+                CancellationToken.None);
+
+            return Task.FromResult(new EvaluateResponseBody
+            {
+                Result = "",
+                VariablesReference = 0
+            });
+        }
```

Comment:
@TylerLeonhardt @SeeminglyScience @rkeithhill in case any of you know why we don't await the result from the evaluate handler... I have an inkling it was due to some kind of deadlock issue?

Reply:
I've had an idea but want to separate out that work -- I'll open an issue